CONTRIBUTING.md

```diff
@@ -1,3 +1,71 @@
 # Contributing to Modus
 
-TODO
+We're really glad you're here and would love for you to contribute to Modus! There are a variety of ways to make Modus better, including bug fixes, features, docs, and blog posts (among others). Every bit helps 🙏
+
+Please help us keep the community safe while working on the project by upholding our [Code of Conduct](/CODE_OF_CONDUCT.md) at all times.
+
+Before jumping to a pull request, ensure you've looked at [PRs](https://github.com/hypermodeinc/modus/pulls) and [issues](https://github.com/hypermodeinc/modus/issues) (open and closed) for existing work related to your idea.
+
+If in doubt or contemplating a larger change, join the [Hypermode Discord](https://discord.hypermode.com) and start a discussion in the [#modus](https://discord.com/channels/1267579648657850441/1292948253796466730) channel.
+
+## Codebase
+
+The primary development language of Modus is Go. The **runtime** and supporting packages are all written in Go.
+
+**Modus SDKs** enable users to build apps in multiple languages, currently including Go and AssemblyScript.
+
+The **Modus CLI** drives the local development experience and is written in TypeScript.
+
+### Development environment
+
+The fastest path to setting up a development environment for Modus is through VS Code. The repo includes a set of configs to set VS Code up automatically.
+
+### Clone the Modus repository
+
+To contribute code, start by cloning the Modus repository,
+
+```bash
+git clone https://github.com/hypermodeinc/modus.git
+```
+
+and create a branch for your work.
+
+```bash
+cd modus
+git checkout -b my-new-branch
+```
+
+### Building and running tests
```

Comment:
This is OK for now, but we'll probably want to describe the monorepo structure and what projects can be built from where.  I can add that later if you want.

Reply:
that would be great! agreed it can come later